include/posix/pthread.h

```diff
@@ -383,4 +385,9 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Sleep Related APIs */
+unsigned int sleep(unsigned int seconds);
+unsigned int msleep(unsigned int mseconds);
```

Comment:
There's no msleep() in POSIX. Convenience functions are fine, but should be separated from true POSIX functions and marked as such.

Reply:
Hi @pfalcon 

Yes as per Posix IEEE Std 1003.13-2003 there is no msleep. But I have put it to provide flexibility to application developer to give delay delay in terms of msecs. Without msleep Application developer will have only usleep as a way to provide delay. So technically on every msec delay we will have busy wait. I wanted to avoid this waste of power at that point.  

Interestingly POSIX standard for linux has msleep.

Reply:
The purpose of the POSIX compatibility is to help porting applications and/or libraries already written -- providing new functions will add very little, if any, value.  It's fine to introduce helper functions if they're used as implementation detail, but I'd rather see those as `static inline`.

Reply:
@youvedeep :

> Interestingly POSIX standard for linux has msleep.

Well, there's no "POSIX standard for linux". There's only one POSIX. Then there cab be vendor extensions, but the care should be taken to not compromise POSIX compliance with them. For example, as POSIX doesn't define msleep(), a POSIX-compliant application may define such a function itself. But if a header which defines standard POSIX functions, like sleep()/usleep(), also defines extensions, like msleep(), they may conflict with a perfectly compliant POSIX applications, like an example given above.

Overall, my point is that there's more to POSIX compliance than just implementing functions defined by POSIX: there should be proper POSIX headers, and they should take care into account namespacing and naming conflict issues like above.

Reply:
> Without msleep Application developer will have only usleep as a way to provide delay. So technically on every msec delay we will have busy wait.

We doesn't have to. It's just POSIX usleep() can be implemented in a smarter way, e.g. for delays below 1000us it can use busy-wait, and for longer delays - non busy-wait implementation.

Reply:
@pfalcon i was planning to do the same :).  
I will remove msleep and make usleep more smart so that it avoids busy wait if delay is more than 1msec. But this may create some confusion as use may feel its behavior is different when she is giving time more than 1msec Vs less than 1msec. We need to document it in order to avoid confusion.